crucible-mir/src/Mir/TransTy.hs

```diff
@@ -1011,6 +1003,50 @@ setEnumField adt i j (MirExp enumTpr e) (MirExp fldTpr e') = do
         " to have type " ++ show (fieldDataType fld) ++ ", but got " ++ show fldTpr
 
 
+-- | A description of an individual field of a union.
+data UnionInfo where
+  UnionInfo ::
+    -- | The overall size of the union
+    Word ->
+    -- | The size of this field
+    Word ->
+    -- | The offset of this field
+    Word ->
```

Comment:
All of the `MirAggregate` ops that take a field's layout info use the opposite ordering: `off` first, then `sz`.